### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by jwerle**
- Assigns `libsodium` export to `crypto.sodium` export 
- Falls back to `sodium.randombytes_buf` in `crypto.getRandomValues`
- Exports `CustomEvent` in `socket:events` for usage in platforms that do not support it globally (node)
- Allow `encoding` to be given in `util.toBuffer()` where `Buffer.from()` is used
- Use `value` as `this` value when calling custom inspect (`Symbol.for('socket.util.inspect.custom')`)
- Introduce `compareBuffers()` for comparing `all `TypedArray` types, including `Buffer`
- Support `--link` for `publish-npm-modules.sh` which globally links the built node modules, instead of publishing  them (this makes it easier to develop against: `npm link @socketsupply/socket` just works)

### CODE REVIEW

**Review by mribbons — APPROVED** [review]

File: api/events.js [hunk 1]
```diff
@@ -469,6 +469,21 @@ function eventTargetAgnosticAddListener (emitter, name, listener, flags) {
 EventEmitter.EventEmitter = EventEmitter
 EventEmitter.once = once
 
+export const CustomEvent = globalThis.CustomEvent || class CustomEvent extends globalThis.Event {
```
Comment (chicoxyzzy) [inline api/events.js:472]: is this for Node.js?

**Review by chicoxyzzy — APPROVED** [review]
it nice to have documentation for npm publishing somewhere (in wiki?)

File: api/events.js (hunk 1 above)
Comment (jwerle) [reply api/events.js:472]: yep!

**Comment by jwerle** [comment]
> it nice to have documentation for npm publishing somewhere (in wiki?)

yeah I agree 🙏🏼  here is some inline until we get a wiki setup on the repo:

```sh
# '--link' will perform a `npm link` and also link the platforms in the top level (it implies '--dry-run' so `npm publish` is not executed
# '--dry-run' does everything except `npm publish`
# '--only-platforms' only publishes (or links) platform modules
# '--only-top-level' only publishes (or links) top level module
publish-npm-modules.sh [--link] [--dry-run] [--only-platforms] [--only-top-level]
```

**Pull request merged**